src/core/thread.d

```diff
@@ -596,7 +596,10 @@ class Thread
      */
     ~this() nothrow @nogc
     {
-        if ( m_addr == m_addr.init )
+        bool no_context = m_addr == m_addr.init;
+        bool not_registered = !next && !prev;
```

Comment:
If there's only one registered thread its `next` and `prev` would also be null, but it appears to me that we don't need to worry about this because it could only happen in the destructor of the main thread.